### DISCUSSION TITLE
MXML / AS Build.xml for ANT

### DISCUSSION THREAD

**Discussion by leossmith** [airsdk/Adobe-Runtime-Support#924] [Q&A]
Can you provide a working build.xml (or at least an example) which can be used to prepare / compile / package apps for android and iOS? With the aab not available in intellij interface this is kinda of important.

I have spent 2-3 hours trying to make a build working, but I keep getting errors with flex. Thank you.

**Comment by marchbold** [comment]
Did you manage to get your build working? We use ant for lots of our projects.

**Comment by ajwfrost** [comment]
Flex build tasks are included in the SDK : see the examples at:
https://joshuarogers.net/articles/2012-05/compiling-adobe-flex-ant/
and there's an example for building Flex apps in the SDK under `frameworks/projects/air/sample-frameworks-build.xml`

For ADT, you'd need to define a task to run Java with the necessary arguments - this is something that could be packaged into an Ant macro to make it a bit clearer perhaps?

What's the error you're getting? There's a known issue with ANEs where the standard "mxmlc" task won't accept them as external libraries (so either they need to be explicitly added as command line params, or they need to be renamed to .swc) - see https://thetopsites.net/article/54927884.shtml

thanks

**Comment by leossmith** [comment]
So I moved on building the ANT script from Intellij to Flash Builder. Intellij provides no information about the errors while FB has a debug option so I know exactly what I am doing. 

@marchbold The actual issue (returning error null) was that manifest <content> not generated in the case of ant. I also used `-dump-config` to get the .xml from my intellij project, instead of using flex-config.xml. 

I managed to move forward and actually compile but, I hit a wall with the ANEs issue @ajwfrost is describing. I will try renaming them to see if that works. Once I am done I will share the project as I have most of the needed tasks already done. 

![image](https://user-images.githubusercontent.com/4300876/123511994-8846b480-d68d-11eb-9736-d12b0e4f9530.png)
